tests/NATS.Client.KeyValueStore.Tests/NatsKVWatcherTest.cs

```diff
@@ -140,6 +140,64 @@ await foreach (var entry in store.WatchAsync<string>("*", cancellationToken: can
         await watchTask;
     }
 
+    [Fact]
+    public async Task Watch_subset()
+    {
+        var timeout = TimeSpan.FromSeconds(10);
+        var cts = new CancellationTokenSource(timeout);
+        var cancellationToken = cts.Token;
+
+        await using var server = NatsServer.StartJS();
+        await using var nats = server.CreateClientConnection();
+
+        var js = new NatsJSContext(nats);
+        var kv = new NatsKVContext(js);
+
+        var bucket = "b1";
+        var store = await kv.CreateStoreAsync(bucket, cancellationToken: cancellationToken);
+
+        await store.PutAsync("k1", "v1", cancellationToken: cancellationToken);
+        await store.PutAsync("k2", "v1", cancellationToken: cancellationToken);
+        await store.PutAsync("k3", "v1", cancellationToken: cancellationToken);
+
+        var signal = new WaitSignal(timeout);
+        var watchTask = Task.Run(
+            async () =>
+            {
+                await foreach (var entry in store.WatchAsync<string>(["k1", "k2"], cancellationToken: cancellationToken))
+                {
+                    signal.Pulse();
+                    _output.WriteLine($"WATCH: {entry.Key} ({entry.Revision}): {entry.Value}");
+
+                    Assert.True(entry.Key is "k1" or "k2");
```

Comment:
we're not checking a couple of cases here:
1) we should see k1
2) we should not see k3

note we can also break on `entry.Delta == 0` to end the watch if that helps.

Reply:
I originally had it as `! "k3"`, but thought `"k1" or "k2"` read better. I did end up adding "k1" after a commit, but it looks like it's being checked here.

Will look at this again though.

Reply:
Looks like I'll need to do more than just copy/paste and rework that watcher test. Seems to be a new exception for 2.9 now.

Reply:
I think we can ignore this test for 2.9:
```csharp
[SkipIfNatsServer(versionEarlierThan: "2.10")]
public async Task Watch_subset()
```

Reply:
just tweaked the test a little to make sure we received both k1 and k2.